test/vim_interface.py

```diff
@@ -85,18 +86,25 @@ def vim_executable(self):
     def has_version(self, major, minor, patchlevel):
         if self._version is None:
             output = subprocess.check_output([self._vim_executable, "--version"])
```

Comment:
What about using Vim's own logic to detect if a patch is included?  You could do something like

```python
output = subprocess.check_output([self._vim_executable, '--cmd', f'echo has("patch-{major}.{minor}.{patchlevel}")', '--cmd', 'quit'])
```

assuming the minimum supported vim version is >= 7.4.237.

Or if it's easier to rely on the exit code

```python
has_version = subprocess.call([self._vim_executable, '--cmd', f'if has("patch-{major}.{minor}.{patchlevel}") | quit | else | cquit | endif'])
```

`:help has-patch` describes an alternative that works for older Vim versions.